CMSgov/design-system
# [WNMGDS-2604] Fix content with stacking contexts bleeding through Healthcare header menu

## Summary

Not really sure _why_ the `isolation: isolate` didn't result in a proper stacking context that could compete with other stacking contexts in the page, but setting `position: relative` instead seemed to do the trick. The added z-index property was to make sure we addressed the problem that `isolation: isolate` was [originally trying to solve](https://github.com/CMSgov/design-system/pull/1216), which was to make sure the header's box shadow rendered over other things on the page.

## How to test

1. Check out d8111de7a329111a70eabbe14387e93796622762 and view the storybook story at the xs breakpoint to see the problem
2. Check out 2be7e3e06b084340181b83fe5db173cb7b2ffe6d to see the fix

<details>
  <summary>Screen capture</summary>
  
  ![Screen capture showing that radio buttons no longer bleed through the menu](https://github.com/CMSgov/design-system/assets/7595652/44befe2b-431e-414c-881e-3fe026caaa97)
</details>


## Checklist




### If this is a change to code:
